compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/invalid-jsx-lowercase-localvar.jsx

```diff
@@ -0,0 +1,29 @@
+import {Throw} from 'shared-runtime';
+
+/**
+ * Note: this is disabled in the evaluator due to different devmode errors.
+ * Found differences in evaluator results
+ *  Non-forget (expected):
+ *  (kind: ok) <invalidtag val="[object Object]"></invalidtag>
+ *  logs: ['Warning: <%s /> is using incorrect casing. Use PascalCase for React components, or lowercase for HTML elements.%s','invalidTag']
+ *
+ *  Forget:
+ *  (kind: ok) <invalidtag val="[object Object]"></invalidtag>
+ *  logs: [
+ *   'Warning: <%s /> is using incorrect casing. Use PascalCase for React components, or lowercase for HTML elements.%s','invalidTag',
+ *   'Warning: The tag <%s> is unrecognized in this browser. If you meant to render a React component, start its name with an uppercase letter.%s','invalidTag',
+ *  ]
+ */
+function useFoo() {
+  const invalidTag = Throw;
```

Comment:
Hmm, can we add validation for this case? It would be better to reject it then to fail silently.

Reply:
@josephsavona Hmm I don't think we need to bail out on this. We currently do lower it correctly (i.e. treat the lowercase jsx tag as a string) and keep the same program semantics. I just added the test fixture for completeness.

I can definitely see the value for a `InvalidReact` lint, without bailing out the compilation pipeline. We can do this by extending `logDiagnostics`. Happy to add this later in a followup

Reply:
(Note that this PR doesn't change behavior for this fixture)